stan/math/prim/mat/fun/gp_exp_quad_cov.hpp

```diff
@@ -177,50 +180,61 @@ gp_exp_quad_cov(const std::vector<T_x1>& x1, const std::vector<T_x2>& x2,
  *
  * @tparam T_x1 type of first std::vector of elements
  * @tparam T_x2 type of second std::vector of elements
- * @tparam T_sigma type of sigma
+ * @tparam T_s type of sigma
  * @tparam T_l type of length scale
  *
- * @param x1 std::vector of elements that can be used in square distance
- * @param x2 std::vector of elements that can be used in square distance
+ * @param x1 std::vector of Eigen column vectors of scalars.
+ * @param x2 std::vector of Eigen column vectors of scalars.
  * @param sigma standard deviation
  * @param length_scale std::vector of length scale
+ *    This function assumes the column vectors of x1 and x2
```

Comment:
Do we need to assume this? I would much prefer to have in the code a quick check. I can see a user getting easily confused and using the wrong size... I think the code as written will kick out and generate the appropriate error, but it'll have the wrong error message (the message will come from something like `squared_distance` which will just confuse the user).

I'll mark exactly where I think this check should go below with a `(*)`.